core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -9633,6 +9641,32 @@ private boolean isUserStoreChainNeeded(List<String> userStorePreferenceOrder) th
     private boolean generateUserStoreChain(String userName, Object credential, boolean domainProvided,
                                            List<String> userStorePreferenceOrder) throws UserStoreException {
 
+        // If domain name is provided, directly authenticate using the corresponding user store.
+        if (domainProvided) {
+            String providedDomainName = this.getUserStore(userName).getDomainName();
+            // Check whether provided domain is in the preference list.
+            if (!userStorePreferenceOrder.contains(providedDomainName)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Authentication failure. Wrong username or password is provided.");
```

Comment:
Shall we change this is `invalid`? WDYT?

Reply:
Done